src/Maintainer.php

```diff
@@ -57,9 +59,14 @@ protected function _rescheduleCrashedJobs(): Maintainer
             $jobSemaphoreResource = $this->_getNewJobOwnerResource($job);
             try {
                 if ($jobSemaphoreResource->testAndSetLock()) {
-                    $crashUpdate = $this->_getServiceUpdateCrashFactory()->create();
-                    $crashUpdate->setJob($job);
-                    $crashUpdate->save();
+                    $mutuallyExcludedJob = $this->_getJobClone();
+                    $mutuallyExcludedJob->setId($job->getId());
+                    $mutuallyExcludedJob->load();
+                    if ($mutuallyExcludedJob->getAssignedState() === State\Service::STATE_WORKING) {
```

Comment:
Would the assigned state of the crashed job not be `STATE_CRASHED`?

Reply:
if the state of the `Job` is *still* working and the maintainer can obtain the mutex for the `Job`, then the worker that was working the `Job` has actually crashed.

Reply:
Ah, that makes sense, thanks